Big bag o' tweaks

WickyNilliams opened the pull request:
This PR contains a host of tweaks, improvements, fixes, file size savings 🎉 

In terms of file size, we see the following changes:

|            | v0.9.4 (kb)  | This PR (kb) | Savings (%) |
|------------|-------------|------------|-------------|
| Unminified | 15.28       | 10.35      | 32          |
| Minified   | 5.76        | 4.26       | 26          |
| Min+GZIP   | 1.7         | 1.38       | 19          |

Pretty good!

WickyNilliams commented on src/Headroom.js:
```diff
@@ -95,378 +14,206 @@ function normalizeTolerance(t) {
+            self.scroller,
+            self.update.bind(self)
+          );
+        },
```
I think I'm going to drop this asynchronous behaviour in a future version. I would have done it now, but I wanted to avoid breaking changes. Developers can always defer initialisation themselves if it matters to them.

hacknug replied:
_potentially restoring previous scroll position_ makes it sound like it never was supposed to be a bulletproof solution so I guess it's okay. Maybe we could add an example to the docs/readme/changelog showing how to deal with it on the new version?

WickyNilliams replied:
Yeah, it was never perfect. A browser may take longer than 100ms to restore scroll position on a slow connection or long page. An addition to the docs sounds like a good idea 👍

hacknug commented:
Completely unnecessary but do you have the numbers for Brotli compression too?

WickyNilliams commented:
Good call, I updated the rollup config to output brotli size :) Here's the results pulled from the travis build:

```
src/Headroom.js → dist/headroom.min.js...
┌───────────────────────────────────────┐
│                                       │
│   Destination: dist/headroom.min.js   │
│   Bundle Size:  4.26 KB               │
│   Gzipped Size:  1.38 KB              │
│   Brotli size: 1.32 KB                │
│                                       │
└───────────────────────────────────────┘
```

WickyNilliams commented:
I rebased this branch and dropped the changes to the jQuery/angular plugins, as they were breaking changes. They can be dealt with in a future release

hacknug commented:
@WickyNilliams let me know if there's anything I can help you with!

WickyNilliams commented:
@hacknug sure thing. I don't think there's much can be done for this release, but maybe you'd like to take a stab at removing the async behaviour we discussed above?
